vignettes/data-combined.Rmd

```diff
@@ -0,0 +1,300 @@
+
+```{r}
+myDataCombined$toDataFrame()
+```
+
+This function returns a [tibble data frame](https://r4ds.had.co.nz/tibbles.html#tibbles-vs.-data.frame). If you wish to modify how it is printed, you can have a look at the available options [here](https://pillar.r-lib.org/reference/pillar_options.html). In fact, let's change a few options and print the data frame again.
+
+```{r, eval=FALSE}
+options(
+  pillar.width = Inf, # show all columns
+  pillar.min_chars = Inf # to turn off truncation of column titles
+)
+
+myDataCombined$toDataFrame()
+```
+
+
+```{r, echo=FALSE}
+# change these settings only temporarily
+withr::with_options(
+  list(
+    pillar.width = Inf, # show all columns
+    pillar.min_chars = Inf # to turn off truncation of column titles
+  ),
+  code = {
+    print(myDataCombined$toDataFrame())
+  }
+)
+```
+
+As can be seen, this data frame attemtps to provide a one-stop access to all important details about the datasets included in the object, including important meta-data.
+
+`{ospsuite}` also provides a few helper functions to modify the data frame further.
+
+When multiple (observed and/or simulated) datasets are present in a data frame, they are likely to have different units. `convertUnits()` function helps to convert them to a common unit^[Note that if you are using `DataCombined` object for plotting functions, you don't need to this conversion; the plotting functions will take care of this internally.].
+
+```{r}
+convertUnits(
+  myDataCombined,
+  xUnit = ospUnits$Time$s,
+  yUnit = ospUnits$`Concentration [mass]`$`µg/l`
+)
+```
+
+## Further functionalities
+
+Grouping of simulated results with observed data inside `DataCombined` also allows 
+to calculate the error between simulated results and observations using the 
+`calculateResiduals()` function. The function calculates the distance between 
+each observed data point within the group and the simulated results. Simulation 
+results interpolated if no simulated value is available for a certain observed 
+time point.
+
+There are two ways to calculate the distance (residuals) - linear and logarithmic - 
+specified by the `scaling` argument. For linear scaling (`scaling = tlf::Scaling$lin`):
+
+> Residuals are calculated as: Simulation value - Observed value. This means that the residuals are defined by absolute differences.
+
+For logarithmic scaling (`scaling = tlf::Scaling$log`):
+
+> Residuals are calculated as: log(Simulation value) - log(Observed value) = log (Simulation Value / Observed Value). This means that the ratio of values is considered which is independent of the magnitude of the value. But for very small observed values, in particular close to 0 values, this can lead to problems, because log(10E-N) = -N can becomes large.
+
+
+```{r}
+#Linear residuals
+calculateResiduals(myDataCombined, scaling = tlf::Scaling$lin)$residualValues
+
+#Logarithmic residuals
+calculateResiduals(myDataCombined, scaling = tlf::Scaling$log)$residualValues
```

Comment:
Can we also calculate residuals within a group? OR only for the whole dataset?

Reply:
Also: if I have more than 1 simulated output within a group: how are residuals calculated then?
E.g. I have in my group {Simulated1, Simulated2, Observed1, Observed2}.
Are then distances between each observed and each simulated calculated or how is it done?

Reply:
> Can we also calculate residuals within a group? OR only for the whole dataset?

No, only for the whole `DataCombined`.

> Also: if I have more than 1 simulated output within a group: how are residuals calculated then?

I guess this has not been specified properly... It should be "distances between each observed and each simulated are calculated", but currently it results in an error... Creating an issue for that.

Reply:
https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1133